### PR TITLE
test(radio-button): added unit test

### PR DESCRIPTION
**Pull request by theJohnnyMe**
**Describe pull-request**  
Added unit test for Radio Button component

**Solving issue**  
Fixes: [CDEP-3041](https://tegel.atlassian.net/browse/CDEP-3041)

**How to test**  
1. Check the test case spreadsheet for[ Radio button](https://scaniaazureservices-my.sharepoint.com/:x:/r/personal/tatyana_bumazhkina_scania_com/Documents/Unit%20test%20scenarios.xlsx?d=wca48bf6d92cf4e8c8f456337e2002ec4&csf=1&web=1&e=P3axMi&nav=MTVfezlCNkZBRkM3LTQ5ODMtNDgzMi04REY4LUY0MjUwNTFGRjE1MH0).
2. Check changes done in test files if they reflect upper specs.
3. Run the branch on your local to see if all tests pass

**Checklist before submission**
- [x] I have added unit tests for my changes (if applicable)
- [x] All existing tests pass
- [ ] I have updated the documentation (if applicable)

[CDEP-3041]: https://tegel.atlassian.net/browse/CDEP-3041?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/core/src/components/radio-button/test/disabled/radio-button.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  });
+
+  test('Hover cursor over Label text 2 radio button -> cursor should NOT be able to click on it', async ({
+    page,
+  }) => {
+    const inputElement = page.locator('input[id="option-2"]');
+    await inputElement.hover();
+    await expect(inputElement).toHaveCSS('cursor', 'not-allowed');
+
+    const labelElement = page.locator('text=Label text 1');
```
Comment (nathalielindqvist) [inline packages/core/src/components/radio-button/test/disabled/radio-button.e2e.ts:51]:
```suggestion
    const labelElement = page.locator('text=Label text 2');
```
Is this supposed to be the label text for the second option?

Comment (theJohnnyMe) [reply packages/core/src/components/radio-button/test/disabled/radio-button.e2e.ts:51]: Good catch! You are right ;)

**Review by nathalielindqvist — APPROVED** [review]
Nice! 🎉

File: packages/core/src/components/radio-button/test/disabled/radio-button.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    await expect(radioButton2).toBeVisible();
+    await expect(page).toHaveScreenshot({ maxDiffPixelRatio: 0 });
+  });
+
+  test('1st radio button is disabled', async ({ page }) => {
+    const disabledLabel1RadioButton = page.locator('input[id="option-1"]:disabled');
+    await expect(disabledLabel1RadioButton).toHaveCount(1);
+  });
+
+  test('1st radio button is button is checked', async ({ page }) => {
```
Comment (timrombergjakobsson) [inline packages/core/src/components/radio-button/test/disabled/radio-button.e2e.ts:24]:
```suggestion
  test('1st radio button is checked', async ({ page }) => {
```

File: packages/core/src/components/radio-button/test/disabled/radio-button.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import { expect } from '@playwright/test';
+import { test } from 'stencil-playwright';
+
+test.describe('Radio button - disabled state', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto('src/components/radio-button/test/disabled/index.html');
+  });
+
+  test('Radio buttons with Label text = "Label text 1" and "Label text 2" render on the page', async ({
```
Comment (timrombergjakobsson) [inline packages/core/src/components/radio-button/test/disabled/radio-button.e2e.ts:9]:
```suggestion
  test('Radio buttons with Label text = "Label text 1" and "Label text 2" renders on the page', async ({
```

**Review by timrombergjakobsson — APPROVED** [review]
Some minor comments otherwise looks great!

**Pull request merged**